presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -211,15 +211,22 @@ private SubPlan reassignPartitioningHandleIfNecessaryHelper(Session session, Sub
 
         private final Session session;
         private final Metadata metadata;
-        private final TypeProvider types;
+        private final PlanNodeIdAllocator idAllocator;
```

Comment:
this looks unused

Reply:
It is used in the next commit. Just wanted to have all the trivial operations in separate commits to simplify reading. I will squash everything before the merge.